setup.py

```diff
@@ -20,7 +20,7 @@
         "dev": [
             "aiomysql",
             "mypy",
-            "mysql-connector-python",
+            "mysql-connector-python<9.0.0",
```

Comment:
I believe there is a behavior change for the default collation in 9.0.0 

Was trying to fix it in https://github.com/kelsin/mysql-mimic/pull/60/files/cd22920bbe455bc711d6c601d80fc17c40e5a2f9 but failed (it can work in python >= 3.9 but failed in python 3.7, probably because it is using mysql-connector-python 8.0 for python 3.7 but 9.0 for python 3.11

Reply:
Another issue combined with the above is [sqlalchemy](https://github.com/sqlalchemy/sqlalchemy/discussions/7858) set `SET NAMES 'utf8mb4' ` on connection, which removed that COLLATE which means we need special handling in testing for sqlalchemy.

So, probably limit mysql-connector is the best way to resolve the issue 🫤

Reply:
We should avoid limiting mysql-connector. It implies mysql-mimic doesn't work with the latest version of mysql-connector.

Can we just change the expected collation in tests?
Or change the collation when connecting in tests?
https://dev.mysql.com/doc/connector-python/en/connector-python-connectargs.html
https://github.com/kelsin/mysql-mimic/blob/main/tests/conftest.py#L210
And I think you can set connect args in the sqlalchemy URL

Reply:
I see. 

Going to run the set names/collate query on each of the connection